libpod/oci_conmon_linux.go

```diff
@@ -783,13 +783,31 @@ func (r *ConmonOCIRuntime) ExecContainer(c *Container, sessionID string, options
 	return pid, attachChan, err
 }
 
+// isRetryable returns whether the error was caused by a blocked syscall or the
+// specified operation on a non blocking file descriptor wasn't ready for completion.
+func isRetryable(err error) bool {
+	if errno, isErrno := errors.Cause(err).(syscall.Errno); isErrno {
+		return errno == syscall.EINTR || errno == syscall.EAGAIN
+	}
+	return false
+}
+
 // ExecAttachResize resizes the TTY of the given exec session.
 func (r *ConmonOCIRuntime) ExecAttachResize(ctr *Container, sessionID string, newSize remotecommand.TerminalSize) error {
 	// TODO: probably want a dedicated function to get ctl file path?
 	controlPath := filepath.Join(ctr.execBundlePath(sessionID), "ctl")
-	controlFile, err := os.OpenFile(controlPath, unix.O_WRONLY, 0)
-	if err != nil {
-		return errors.Wrapf(err, "could not open ctl file for terminal resize for container %s exec session %s", ctr.ID(), sessionID)
+	var controlFile *os.File
+	var err error
+
+	for {
```

Comment:
Should we build in a max wait for this loop? 10 seconds or so?

Reply:
not sure, the control file will eventually be deleted and the loop returns.  But I can add one.  Should I hard-code 10 seconds or play safer with something like 60 seconds?

Reply:
60 seconds,   Will this go into a tight loop if the failure happens?

Reply:
I like the idea of stopping after X attempts.